python/pyspark/cloudpickle.py

```diff
@@ -1019,18 +948,40 @@ def __reduce__(cls):
         return cls.__name__
 
 
-def _fill_function(func, globals, defaults, dict, module, closure_values):
-    """ Fills in the rest of function data into the skeleton function object
-        that were created via _make_skel_func().
+def _fill_function(*args):
+    """Fills in the rest of function data into the skeleton function object
+
+    The skeleton itself is create by _make_skel_func().
```

Comment:
Restore compatibility with functions pickled with 0.4.0 (#128) 
https://github.com/cloudpipe/cloudpickle/commit/7d8c670b703a683d6fd7e642c6bec8a487594d20

Reply:
Yea, I think we need this.

Reply:
That's more of an issue if using pickles stored on disk or if nodes in the cluster are on different versions. Is that likely for Spark?

Reply:
I don't think we made the guarantee on it but the best is always to stay safer. Others seem bug fixes but this one could be a regression (about the support we haven't guaranteed). It's the part of 0.4.2v anyway.